primitives/teerex/src/lib.rs

```diff
@@ -52,3 +56,26 @@ pub struct Request {
 	pub shard: ShardIdentifier,
 	pub cyphertext: Vec<u8>,
 }
+
+#[derive(PartialEq, Eq, Clone, Encode, Decode, Debug, Copy, Default, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct Header {
```

Comment:
```suggestion
pub struct SidechainBlockHeader {
```
Otherwise, things might get very confusing

Reply:
the plan is to extract this to a sidechain specific pallet (https://github.com/integritee-network/pallets/issues/49, in progress). This also resolves the confusion.

Reply:
Even in its own pallet, it should be named `SidechainBlockHeader` IMO - it's not the pallet that creates the ambiguity, but the fact that we're in a parentchain context (which will not change with a different pallet)

Reply:
ok

Reply:
But why SidechainBlockHeader and not SidechainHeader? Is it not obvious that it is the header of a block?